Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -3047,6 +3069,305 @@ def remove_duplicates_from_list_arg(args: dict, field: str):
+            'interval_in_seconds': interval_in_secs,
+            'polling': True,
+            **args,
+        }
+        scheduled_command = ScheduledCommand(
+            command=cmd,
+            next_run_in_seconds=interval_in_secs,
+            args=polling_args,
+            timeout_in_seconds=timeout_in_seconds)
+        command_results.scheduled_command = scheduled_command
+        return command_results
+
+    # not a first run
+
+    command_result = results_function(client, args)
+    action_status = command_result.outputs.get("status")
+    command_status = command_result.outputs.get("commands", [{}])[0].get("commandStatus")
+    if action_status in ['Failed', 'Cancelled'] or command_status == 'Failed':
+        raise Exception(
+            f'Command {action_status}. Additional info: {command_result.outputs.get("commands", [{}])[0].get("errors")}')
+    elif command_status != 'Completed' or action_status == 'InProgress':
+        # schedule next poll
+        polling_args = {
+            'interval_in_seconds': interval_in_secs,
+            'polling': True,
+            **args
+        }
+
+        scheduled_command = ScheduledCommand(
+            command=cmd,
+            next_run_in_seconds=interval_in_secs,
+            args=polling_args,
+            timeout_in_seconds=timeout_in_seconds
+        )
+
+        command_result = CommandResults(scheduled_command=scheduled_command)
+        return command_result
+
+    # action was completed
+    else:
+        return post_polling_process(client, command_result.outputs)
+
+
+def get_live_response_result_command(client, args):
+    machine_action_id = args['machine_action_id']
+    command_index = arg_to_number(args['command_index'])
+    res = client.get_live_response_result(machine_action_id, command_index)
+    file_link = res['value']
+
+    # download link, create file result
+    f_data = client.download_file(file_link)
+    try:
+        outputs = f_data.json()
+    except Exception:
+        outputs = {'value': file_link}
+
+    return [fileResult('Response Result', f_data.content), CommandResults(
+        outputs_prefix='MicrosoftATP.LiveResponseResult',
+        outputs=outputs,
+        readable_output=f'file_link: {file_link}'
+    )]
+
+
+def get_machine_action_command(client, args):
+    id = args['machine_action_id']
+    res = client.get_machine_action_by_id(id)
+
+    return CommandResults(
+        outputs_prefix='MicrosoftATP.MachineAction',
+        outputs_key_field='action_id',
+        outputs=res
+    )
+
+
+def cancel_action_command(client, args):
+    action_id = args['machine_action_id']
+    comment = args['comment']
+    body = {
+        "Comment": comment
+    }
+    # cancel action should return either 200 or 404.
+    try:
+        client.cancel_action(action_id, body)
+    except Exception as e:
+        if '404' in str(e):
+            raise Exception(f'Action ID {action_id} could not be found. Make sure you entered the correct ID.')
```

Comment:
Not critical, but in general it's a bad practice to raise `Exception`, it's better to raise it as a specific class such as `DemistoException`.